apps/src/Register3D.cpp

```diff
@@ -0,0 +1,185 @@
+        imageFixed = reader.getTextureMat();
+        fixedImage =
+            itk::OpenCVImageBridge::CVMatToITKImage<rt::Image8UC3>(imageFixed);
+    } catch (volcart::IOException& excp) {
+        std::cerr << "Exception thrown " << std::endl;
+        std::cerr << excp.what() << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    auto cvMovingImage = cv::imread(movingImageFileName);
+    // Read the two images
+    movingImage =
+        itk::OpenCVImageBridge::CVMatToITKImage<rt::Image8UC3>(cvMovingImage);
+
+    // Ignore spacing information
+    fixedImage->SetSpacing(1.0);
+    movingImage->SetSpacing(1.0);
+
+    // Read the landmarks file
+    rt::LandmarkIO landmarkReader(landmarksFileName);
+    landmarkReader.setFixedImage(fixedImage);
+    landmarkReader.setMovingImage(movingImage);
+    landmarkReader.read();
+    auto fixedLandmarks = landmarkReader.getFixedLandmarks();
+    auto movingLandmarks = landmarkReader.getMovingLandmarks();
+
+    ///// Landmark Registration /////
+    printf("Running landmark registration...\n");
+
+    // Generate the landmark transform
+    rt::LandmarkRegistration landmark;
+    landmark.setFixedLandmarks(fixedLandmarks);
+    landmark.setMovingLandmarks(movingLandmarks);
+    auto ldmTransform = landmark.compute();
+
+    // Apply it to the image
+    auto tmpMovingImage =
+        rt::ImageTransformResampler(fixedImage, movingImage, ldmTransform);
+
+    ///// Deformable Registration /////
+    printf("Running deformable registration...\n");
+    rt::DeformableRegistration deformable;
+    deformable.setFixedImage(fixedImage);
+    deformable.setMovingImage(tmpMovingImage);
+    auto deformTransform = deformable.compute();
+
+    ////// USED FOR READING IN TRANSFORM FILE //////
+    /*TransformReader::Pointer transformReader = TransformReader::New();
+    transformReader->SetFileName(transformIn);
+    try {
+        // Software Guide : BeginCodeSnippet
+        transformReader->Update();
+    }
+    catch(itk::ExceptionObject & excp) {
+        std::cerr << "Error while reading the transform file" << std::endl;
+        std::cerr << excp << std::endl;
+        std::cerr << "[FAILED]" << std::endl;
+        return EXIT_FAILURE;
+    }
+    const TransformReader::TransformListType *transforms =
+    transformReader->GetTransformList();
+
+    CompositeTransform::Pointer compositeTrans;
+    TransformReader::TransformListType::const_iterator it = transforms->begin();
+    if(!strcmp((*it)->GetNameOfClass(),"CompositeTransform")) {
+        compositeTrans = static_cast< CompositeTransform* >((*it).GetPointer());
+        compositeTrans->Print(std::cout);
+    }*/
+
+    ///// Combine transforms /////
+    CompositeTransform::Pointer compositeTrans = CompositeTransform::New();
+    compositeTrans->AddTransform(ldmTransform);
+    compositeTrans->AddTransform(deformTransform);
+
+    printf("Finished registration\n\n");
+    auto origUVMap = reader.getUVMap();
+    auto newUVMap = volcart::UVMap();
```

Comment:
You probably only need one UVMap since you're overwriting the values at each index (and each index doesn't depend on any other).